src/test/jtx/impl/WSClient.cpp

```diff
@@ -205,6 +205,121 @@ class WSClientImpl : public WSClient
         return {};
     }
 
+    Json::Value
+    invoke(Json::Value const& cmd) override
+    {
+        using boost::asio::buffer;
+        using namespace std::chrono_literals;
+        Json::Value request;
+        if (cmd.isArray())
+        {
+            auto const num_requests = cmd.size();
+            for (unsigned i = 0; i < num_requests; ++i)
+            {
+                Json::Value const& cmdi = cmd[i];
+
+                if (cmdi.isMember(jss::params))
+                {
+                    auto const& params = cmdi[jss::params][0u];
+                    for (auto j = params.begin(); j != params.end(); ++j)
+                        request[i][j.key().asString()] = *j;
+                }
+                for (auto j = cmdi.begin(); j != cmdi.end(); ++j)
+                {
+                    if (j.key().asString() != "params")
+                        request[i][j.key().asString()] = *j;
+                }
+            }
+        }
+        else
+        {
+            if (cmd.isMember(jss::params))
+            {
+                auto const& params = cmd[jss::params][0u];
+                for (auto i = params.begin(); i != params.end(); ++i)
+                    request[i.key().asString()] = *i;
+            }
+            for (auto i = cmd.begin(); i != cmd.end(); ++i)
+            {
+                if (i.key().asString() != "params")
+                    request[i.key().asString()] = *i;
+            }
+        }
+
+        {
+            auto const s = to_string(request);
+            ws_.write_frame(true, buffer(s));
+        }
+
+        Json::Value response;
+        if (cmd.isArray())
+        {
+            for (unsigned i = 0; i < cmd.size(); ++i)
+            {
+                auto jv = findMsg(5s,
```

Comment:
Its not possible for the server to send out responses as soon as it creates them, because the JSON-RPC 2.0 specification stipulates that the responses to a batch request are represented by a single top-level JSON array, containing one element for each response (in an unspecified order).